src/clients/Device.js

```diff
@@ -766,55 +613,69 @@ class Device extends EventEmitter {
     return await this.listenFor('VariableValue', null, token);
   };
 
-  _callFunction = async (
-    name: string,
-    args: Object,
+  // call function on device firmware
+  callFunction = async (
+    functionName: string,
+    functionArguments: Object,
   ): Promise<*> => {
-    try {
-      const buffer = await this._transformArguments(name, args);
-      if (!buffer) {
-        throw new Error(`Unknown Function ${name}`);
-      }
-
-      if (settings.showVerboseDeviceLogs) {
-        logger.log(
-          'sending function call to the core',
-          { deviceID: this._id, name },
-        );
-      }
+    const isBusy = !this._isSocketAvailable(null);
+    if (isBusy) {
+      throw new Error('This core is locked during the flashing process.');
+    }
 
-      const writeUrl = (message: Message): Message => {
-        message.setUri(`f/${name}`);
-        if (buffer) {
-          message.setUriQuery(buffer.toString());
-        }
+    const buffer = await this._transformArguments(
+      functionName,
+      functionArguments,
+    );
 
-        return message;
-      };
+    if (!buffer) {
+      throw new Error(`Unknown Function ${functionName}`);
+    }
 
-      const token = this.sendMessage(
-        'FunctionCall',
-        {
-          _writeCoapUri: writeUrl,
-          args: buffer,
-          name,
-        },
-        null,
+    if (settings.showVerboseDeviceLogs) {
+      logger.log(
+        'sending function call to the core',
+        { deviceID: this._id, functionName },
       );
-
-      const message = await this.listenFor('FunctionReturn', null, token);
-      return this._transformFunctionResult(name, message);
-    } catch (error) {
-      throw error;
     }
+
+    const writeUrl = (message: Message): Message => {
+      message.setUri(`f/${functionName}`);
+      if (buffer) {
+        message.setUriQuery(buffer.toString());
+      }
+
+      return message;
+    };
+
+    const token = this.sendMessage(
+      'FunctionCall',
+      {
+        _writeCoapUri: writeUrl,
+        args: buffer,
+        name: functionName,
+      },
+      null,
+    );
+
+    const message = await this.listenFor('FunctionReturn', null, token);
+    return this._transformFunctionResult(functionName, message);
   };
 
   /**
    * Asks the core to start or stop its 'raise your hand' signal
-   * @param showSignal - whether it should show the signal or not
-   * @param callback - what to call when we're done or timed out...
    */
-  _raiseYourHand = async (showSignal: boolean, ..._:void[]): Promise<*> => {
+    // TODO figure out what is it for and why messageName isn't
+    // consistent with messages specification
+  raiseYourHand = async (
```

Comment:
Thanks @durielz - I think we'll just ask you when we have questions haha. You seem to know everything about the CLI/Firmware